src/SerEvaluator.cs

```diff
@@ -364,7 +364,8 @@ await responseStream.WriteAsync(GetResult(new OnDemandResult()
             foreach (var task in config.Tasks)
                 foreach (var report in task.Reports)
                 {
-                    var connection = report?.Connections?.FirstOrDefault(c => c.App == appId) ?? null;
+                    var vaildConnections = report.Connections.Take(report.Connections.Count - 1);
+                    var connection = vaildConnections?.FirstOrDefault(c => c.App == appId) ?? null;
```

Comment:
please add inline comments, so that  someone can understand this code, because this is special